src/zarr/abc/store.py

```diff
@@ -329,17 +329,17 @@ def supports_listing(self) -> bool:
         ...
 
     @abstractmethod
-    def list(self) -> AsyncGenerator[str, None]:
+    async def list(self) -> AsyncGenerator[str, None]:
         """Retrieve all keys in the store.
 
         Returns
         -------
         AsyncGenerator[str, None]
         """
-        ...
+        yield  # type: ignore[misc]
```

Comment:
Would it be better to yield an empty string to avoid having to silence mypy?

Reply:
Sure, that should work.